vars/buildImage.groovy

```diff
@@ -0,0 +1,99 @@
+// Create a Job and BuildConfig Template to trigger the Build
+// Available parameters:
+//    dockerFile    string -- DockerFile used for the buildconfig
+//    gitURL:       string -- Git URL used for the buildconfig
+//    image         string -- Image used for the buildconfig
+//    nameSpace:    string -- Namespace name to run the build
+def call(params = [:]) {
+    def imageName
+    pipeline {
+        // We don't need any specify agent just to add the buildconfig and trigger it,
+        // this pod will be destroyed in the end
+        agent {
+            node { label 'nodejs' }
+        }
+        stages {
+            stage('Prepare Env') {
+                steps{
+                   script{
```

Comment:
Can we use scripted pipeline instead? We shouldn't have to provision an agent either; let's leave that to callers. So e.g. this would be something like:

```groovy
def call(params = [:]) {
    openshift.withCluster() {
        openshift.withProject() {
            // load JSONs and set their values
            // create objects
            // follow logs
        }
    }
```

I left out the stages. I think it'probably fine without them, but you can include them as stages nested within the `openshift.withProject()` closure if you'd like.